### PR TITLE
NEW: adds ResultCollection to Usage API

### PR DESCRIPTION
**Pull request by lizgehret**
Adds ResultCollections to Usage API with the following steps (notes for myself while developing):

1. Change base Usage class to accept Collections
2. Implement in ExecutionUsage class
3. Use dummy plugin to test that this works

Co-Authored by: @Oddant1

### CODE REVIEW

**Comment by lizgehret** [comment]
Accompanying CLI PR: [#302](https://github.com/qiime2/q2cli/pull/302)

File: qiime2/sdk/usage.py [hunk 1]
```diff
@@ -1608,6 +1664,12 @@ def init_artifact(self, name, factory):
 
         return variable
 
+    def init_result(self, name, factory):
```
Comment (ebolyen) [inline qiime2/sdk/usage.py:1667]: This seems to be the wrong name

**Review by ebolyen — COMMENTED** [review]
This all looks pretty on the level, but we will need a way to assert something about a given `key` in a collection, which will be important for q2-mystery-stew

**Comment by ebolyen** [comment]
For the assertion, I believe we discussed just adding a `key=...` parameter to the assertion methods, that way we don't have to invent a de-referencing operation in the API itself. But if you guys want to make a de-reference operation (turning a `result_collection` variable into an `artifact` variable), I'm not opposed.

**Review by ebolyen — APPROVED** [review]
Need to pull this down, but it looks very reasonable. Nice work!

**Review by lizgehret — COMMENTED** [review]
recent additions look good!

**Pull request merged**